addons/project_timesheet_holidays/tests/test_timesheet_holidays.py

```diff
@@ -243,3 +243,14 @@ def test_timeoff_task_creation_with_holiday_leave(self):
         self.assertEqual(task_count + 1, new_task_count)
         new_timesheet_count = self.env['account.analytic.line'].search_count([('holiday_id', '!=', False)])
         self.assertEqual(timesheet_count + 1, new_timesheet_count)
+
+    def test_non_timesheet_user_access_to_task_form(self):
```

Comment:
I don't think this use-case is particularly useful 
1. It doesn't specify which form view we are using, but rather picks the first form view for this record, which may or may not even have the problem we are testing for to begin with (no ``is_time_off_task`` field)
2. I also think it's a bit too specific of a test case to be useful in general - it's very low probability that this test case will catch a potential bug in the future, and it is code that will have to be maintained

Reply:
> I don't think this use-case is particularly useful
> 
> 1. It doesn't specify which form view we are using, but rather picks the first form view for this record, which may or may not even have the problem we are testing for to begin with (no `is_time_off_task` field)
> 2. I also think it's a bit too specific of a test case to be useful in general - it's very low probability that this test case will catch a potential bug in the future, and it is code that will have to be maintained

We added this field to the inherited view (without primary mode), so it will appear in the default view. Therefore, I did not use a specific view. what do you think?

Reply:
I am mostly arguing that specifying the xml id of the view we are testing is more robust - for example because the view fetched by the Form function could change in the future. A change like that might not get caught by the test in the future and thus go undetected :thinking: 
(Also I'm making the case that imo the test is not really needed because it's very specific (not wider coverage) and it's more tech debt than what it's worth)